### PR TITLE
[Feature] support DEKR

### PR DESCRIPTION
**Pull request by JinluZhang1126**
## Motivation

To support the single-stage model, DEKR, which is missed in previous versions of MMPose.

## Modification

add single-stage config files in the './configs' directory;
add OneStage detector class and corresponding regression head in the './mmpose/model' directory;
update the bottom-up data transformation like generating offsets, multi-scale, and match_to_heatmap in the './mmpose/datasets' directory...

## BC-breaking (Optional)

No.

## Use cases (Optional)
```bash
python tools/train.py configs/body/2d_kpt_sview_rgb_img/onestage/coco/hrnet_w48_coco_640x640.py
```

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: configs/body/2d_kpt_sview_rgb_img/disentangled_keypoint_regression/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Bottom-up Human Pose Estimation via Disentangled Keypoint Regression (DEKR)
```
Comment (ly015) [inline configs/body/2d_kpt_sview_rgb_img/disentangled_keypoint_regression/README.md:1]: I suggest using 'dekr' as the folder name for simplicity.

File: mmpose/datasets/pipelines/bottom_up_transform.py [hunk 2]
```diff
@@ -733,41 +976,13 @@ def __call__(self, results):
             input_size = np.array([input_size, input_size], dtype=np.int)
         img = results['img']
 
-        h, w, _ = img.shape
-
-        # calculate the size for min_scale
-        min_input_w = _ceil_to_multiples_of(self.min_scale * input_size[0], 64)
```
Comment (jin-s13) [inline mmpose/datasets/pipelines/bottom_up_transform.py:739]: please double check this part.

File: mmpose/datasets/pipelines/bottom_up_transform.py [hunk 3]
```diff
@@ -572,38 +708,138 @@ def __call__(self, results):
...
+                                    '`gen_center_heatmap` is True. ' \
+                                    'e.g. sigma=(2, 4)'
+
         self.sigma = sigma
+        self.bg_weight = bg_weight
+        self.gen_center_heatmap = gen_center_heatmap
         self.use_udp = use_udp
 
-    def _generate(self, num_joints, heatmap_size):
+    def _generate(self, num_joints, sigma, heatmap_size):
```
Comment (jin-s13) [inline mmpose/datasets/pipelines/bottom_up_transform.py:745]: is sigma necessary?

Comment (Ben-Louis) [reply mmpose/datasets/pipelines/bottom_up_transform.py:745]: `BottomUpGenerateHeatmapTarget` is modified to support the generation of both keypoint and instance center heatmaps. These two kinds of heatmaps are generally rendered with different sigma. 

Thus, `self.sigma` is changed to a tuple that contains sigmas for keypoint and instance center heatmaps. The parameter `sigma` of function `_generate` is the sigma for either keypoint or instance center heatmaps

File: mmpose/datasets/pipelines/bottom_up_transform.py (hunk 2 above)
Comment (Ben-Louis) [reply mmpose/datasets/pipelines/bottom_up_transform.py:739]: Thanks, the option `use_udp` is lost. After fixing this problem, I compare the deleted codes with the function `_get_multi_scale_size` and find that they match well. On 10000 random inputs, two versions of `BottomUpGetImgSize` generate the same outputs

**Review by ly015 — APPROVED** [review]

**Pull request merged**